### PR TITLE
feat(tekton)!: update tekton UX

### PR DESCRIPTION
**Pull request by divyanshiGupta**
Fixes: https://github.com/janus-idp/backstage-plugins/issues/804

This PR updates tekton plugin UX based on new designs. It also removes support for a separate tekton tab.

Screenshots:

https://github.com/janus-idp/backstage-plugins/assets/20724543/dd157912-f7d8-46fd-8b65-5d638dec6db8

To-Do:

- [x] Update Readme
- [x] Add more unit tests

### CODE REVIEW

**Comment by divyanshiGupta** [comment]
cc: @ShiranHi

**Comment by ShiranHi** [comment]
@divyanshiGupta that looks great!
I have 3 questions:
1. Is it possible to decrease the padding between the arrow used to collapse and expand rows in the table and the first column NAME? The current gap looks quite large.
2. Could we consider including tooltips for when you hover over the "Collapse all" and "Expand all" icons?
2. How about adding an icon for each status in the filter? Let's go with the same icons we use on the table

**Comment by divyanshiGupta** [comment]
> @divyanshiGupta that looks great! I have 3 questions:
> 
> 1. Is it possible to decrease the padding between the arrow used to collapse and expand rows in the table and the first column NAME? The current gap looks quite large.
> 2. Could we consider including tooltips for when you hover over the "Collapse all" and "Expand all" icons?
> 3. How about adding an icon for each status in the filter? Let's go with the same icons we use on the table

@ShiranHi thanks for the feedback. Will try to add the changes that you have suggested.

File: plugins/tekton/src/components/common/TableExpandCollapse.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+});
+
+export const TableExpandCollapse = () => {
+  const classes = useStyles();
+  const { isExpanded, setIsExpanded } = React.useContext(
+    TektonResourcesContext,
+  );
+  // const [collapseDisabled, setCollapseDisabled] =
+  //   React.useState<boolean>(false);
+  // const [expandDisabled, setExpandDisabled] = React.useState<boolean>(false);
```
Comment (debsmita1) [inline plugins/tekton/src/components/common/TableExpandCollapse.tsx:23]: remove these comments

**Review by debsmita1 — CHANGES_REQUESTED** [review]

**Comment by debsmita1** [comment]
The PipelineRunVisualization component doesn't need its own TektonResourcesContext Provider.

**Comment by debsmita1** [comment]
In the dark theme mode, the list looks like this
Screenshot 2023-10-19 at 3 39 04 PM

**Comment by divyanshiGupta** [comment]
@ShiranHi I have reduced padding for expand/collapse buttons and added tooltips as well. Also header table cells have same padding for all but the gap between name and status cell seem more because the plr name occupies more space and hence has more width so the name cell aligns itself with it to look consistent hence increasing the space between the cells.


https://github.com/janus-idp/backstage-plugins/assets/20724543/ad7b4c48-2b3e-4cea-ad8d-8e5579dbe66e


@debsmita1 I have fixed the dark theme issue.

**Comment by divyanshiGupta** [comment]
@ShiranHi currently the MUI core `Select` component currently doesnt support anything apart from strings. There is another component  `Automcomplete` https://v4.mui.com/components/autocomplete/  from MUI lab which supports adding icons but since this component is under lab it might be experimental and is not used anywhere else as of now in the code base. cc: @invincibleJai

**Comment by ShiranHi** [comment]
@divyanshiGupta got it. And thank you for adding the tooltips, looks good to me.

File: plugins/tekton/src/components/PipelineRunList/PipelineRunListSearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  IconButton,
+  Input,
+  InputAdornment,
+  makeStyles,
+} from '@material-ui/core';
+import Clear from '@material-ui/icons/Clear';
+import Search from '@material-ui/icons/Search';
+
+type PipelineRunListSearchBarProps = {
+  pipelineRuns?: any[];
```
Comment (invincibleJai) [inline plugins/tekton/src/components/PipelineRunList/PipelineRunListSearchBar.tsx:15]: nit: can we have type for it?

```suggestion
  pipelineRuns?: PipelineRunKind[];
```

**Review by invincibleJai — COMMENTED** [review]
Thanks @divyanshiGupta, this looks good. Have added a few nits.

**Review by debsmita1 — CHANGES_REQUESTED** [review]

**Comment by invincibleJai** [comment]
@divyanshiGupta I see one edge case with multi-cluster and two warnings 


https://github.com/janus-idp/backstage-plugins/assets/5129024/c1a59ec5-51e0-48f5-ba53-fa7be2051a2c


image

image

**Comment by divyanshiGupta** [comment]
> @divyanshiGupta I see one edge case with multi-cluster and two warnings
> 
>  tektonUX2.mov 
> image image

This issue is not reproducible for me. 

https://github.com/janus-idp/backstage-plugins/assets/20724543/5ad87ef1-9509-4d53-b316-33c3393942f7

Also I dont see the first warning. The warning regarding the key prop is from main.

**Comment by invincibleJai** [comment]
@divyanshiGupta Okay, let's get this and we observe this issue we can take it up an an issue.

**Pull request merged**